projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row-dimension-header.component.ts

```diff
@@ -27,14 +30,68 @@ export class IgxPivotRowDimensionHeaderComponent extends IgxGridHeaderComponent
         public refInstance: ElementRef<HTMLElement>
     ) {
         super(grid, colResizingService, cdr, refInstance);
+        this.pivotGrid = this.grid as PivotGridType;
     }
 
     @HostListener('click', ['$event'])
     public override onClick(event: MouseEvent) {
         event.preventDefault();
     }
 
+    /**
+     * @hidden @internal
+     */
+    public override ngDoCheck() {
+        super.ngDoCheck();
+        this.setSortIndex();
```

Comment:
Won't this trigger too many times (on every check)? Couldn't it be called only when needed, for example when sorting expression change?

Reply:
There's a safety check for that in the method, it can be made with event handler as well.